src/rajawali/Object3D.java

```diff
@@ -649,8 +649,8 @@ protected void cloneTo(Object3D clone, boolean copyMaterial) {
 	public Object3D clone(boolean copyMaterial, boolean cloneChildren) {
 		Object3D clone = new Object3D();
 		cloneTo(clone, copyMaterial);
-		clone.setRotation(getRotation());
-		clone.setScale(getScale());
+        clone.setRotation(getRotation().clone());
+        clone.setScale(getScale().clone());
```

Comment:
This seems to be the same as above, setAll is used so each Vector on each object should be unique.